packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -95,92 +128,115 @@ publishFunctionAppScripts() {
     exitWithUsageInfo
 fi
 
-if [ -z $functionAppNamePrefix ]; then
-    functionAppNamePrefix="$packageName-allyfuncapp"
-fi
-
 if [ -z $clientId ] && [ ! $environment = "dev" ]; then
     echo "AAD application client ID option is required for the non-dev environment."
     exitWithUsageInfo
 fi
 
-# Create AAD function application for dev deployment only if not exists
-if [ -z $clientId ]; then
-    echo "Create AAD application for the development environment if not exists..."
-    functionAppName=$(az group deployment show -g "$resourceGroupName" -n "function-app-template" --query "properties.parameters.name.value" -o tsv 2>/dev/null) || true
-    clientId=$(az webapp auth show -n "$functionAppName" -g "$resourceGroupName" --query "clientId" -o tsv 2>/dev/null) || true
-    appRegistrationName=$(az ad app show --id "$clientId" --query "displayName" -o tsv 2>/dev/null) || true
-
-    if [[ ! -n $appRegistrationName ]]; then
-        createAppRegistration $resourceGroupName $environment
-    else
-        echo "AAD application with display name '$appRegistrationName' already exists."
-    fi
-fi
-
-# Start function app deployment
-echo "Deploying Azure Function App using ARM template..."
-resources=$(az group deployment create \
-    --resource-group "$resourceGroupName" \
-    --template-file "$templateFilePath" \
-    --parameters clientId="$clientId" namePrefix="$functionAppNamePrefix" \
-    --query "properties.outputResources[].id" \
-    -o tsv)
-
-. "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
-functionAppName=$resourceName
-echo "Successfully deployed Azure Function App '$functionAppName'"
-
-# Add reply url to function app registration
-if [ $environment = "dev" ]; then
-    addReplyUrlIfNotExists $clientId $functionAppName
-fi
-
-# Grant key vault access to function app
-echo "Fetching principal ID of the Azure Function App..."
-principalId=$(az functionapp identity show --name $functionAppName --resource-group $resourceGroupName --query "principalId" -o tsv)
-echo "  Successfully fetched principal ID $principalId"
-
-. "${0%/*}/key-vault-enable-msi.sh"
-
-# Start publishing function scripts
-echo "Publishing API functions to '$functionAppName' Function App"
-
-# Install Azure Functions Core Tools for non-dev environment
-osVersion="$(uname -s)"
-case "${unameOut}" in
-Linux*) installAzureFunctionsCoreToolsOnLinux ;;
-*) echo "Azure Functions Core Tools is expected to be installed on development computer. Refer to https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#v2 if tools is not installed." ;;
-esac
-
-# Publish the function scripts to the function app
-publishFunctionAppScripts
+installAzureFunctionsCoreTools
 
-echo "Successfully published API functions to '$functionAppName' Function App."
+deployWebWorkersFunctionApp "web-workers"
+deployWebApiFunctionApp "web-api"
+functionAppName=$functionAppServiceName
```

Comment:
it might be better to explicitly set the value to web worker function here, so that it is obvious. else, at least add comment that "deployWebApiFunctionApp "web-api"" should be above this line.